library/ssl_tls.c

```diff
@@ -1562,7 +1589,8 @@ int mbedtls_ssl_encrypt_buf( mbedtls_ssl_context *ssl,
     mbedtls_cipher_mode_t mode;
     int auth_done = 0;
     unsigned char * data;
-    unsigned char add_data[13];
+    unsigned char add_data[13 + SSL_CID_LEN_MAX ];
```

Comment:
Note: https://tls.mbed.org/kb/how-to/reduce-mbedtls-memory-and-storage-footprint should be expanded at some point with the CID length settings, as they now impact stack usage.